functions/pupil-login/index.js

```diff
@@ -1,8 +1,25 @@
 'use strict'
 
+const v1 = require('./v1')
+const validateMessage = require('./message-validator')
+
 module.exports = async function (context, message) {
   context.log(
     `processing pupil-login message version:${message.version} for check:${message.checkCode}`
   )
-  context.done()
+
+  validateMessage(message)
+
+  switch (message.version) {

```

Comment:
@jon-shipley I think the switch is overcomplicating things here and there's only 1 case.
Why not a simple if?

Reply:
@GeorgeChatzigiannis - this is a versioned message format - this pattern can be extended easily to allow future versions.

If we had an if clause we would still require the else part to cover the default case of a rejecting a message it can't process.